client/state/data-layer/wpcom/me/two-step/index.js

```diff
@@ -0,0 +1,116 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { noop, camelCase } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	TWO_STEP_REQUEST,
+	TWO_STEP_VALIDATE_CODE_REQUEST,
+	TWO_STEP_SEND_SMS_CODE_REQUEST,
+} from 'state/action-types';
+import {
+	setTwoStep,
+	setTwoStepCodeValidationResult,
+	setTwoStepSendSMSValidationCodeResult,
+} from 'state/two-step/actions';
+import { requestUserProfileLinks } from 'state/profile-links/actions';
+import { isReauthRequired } from 'state/two-step/selectors';
+import userSettings from 'lib/user-settings';
+import applicationPasswords from 'lib/application-passwords-data';
+import connectedApplications from 'lib/connected-applications-data';
+
+export const fromApi = data =>
+	Object.keys( data ).reduce( ( accumulator, currentKey ) => {
+		accumulator[ camelCase( currentKey ) ] = data[ currentKey ];
+		return accumulator;
+	}, {} );
+
+export const requestTwoStep = ( { dispatch }, action ) =>
+	dispatch(
+		http(
+			{
+				apiVersion: '1.1',
+				method: 'GET',
+				path: '/me/two-step/',
+			},
+			action
+		)
+	);
+
+/*
+ * Store the fetched user two step confirugration to Redux state
+ */
+export const storeFetchedTwoStep = ( { dispatch }, action, data ) => {
+	dispatch( setTwoStep( fromApi( data ) ) );
+};
+
+export const validateTwoStepCode = ( { dispatch }, action ) =>
+	dispatch(
+		http(
+			{
+				apiVersion: '1.1',
+				method: 'POST',
+				path: '/me/two-step/validate',
+				body: {
+					code: action.code,
+					...( action.action ? { action: action.action } : {} ),
```

Comment:
I don't think those conditional properties are necessary, just pass undefined if they are imo

Reply:
I'm not sure how the backend will handle it, why pass it?